soc: arm: stm32l4 serie low power modes substates

FRASTM opened the pull request:
Change the substate-id range from 1-3 as described in the include/power/power_state.h
this is changing the range in pm_power_state_set() of the soc/arm/st_stm32/stm32l4/power.c
The debug config will let the clocks active in STOP mode at init.The debug config will let the clocks active in STOP mode at init.
Following the #33131, the power.c part of the stm32wb is modified in the same way.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on soc/arm/st_stm32/stm32l4/power.c:
```diff
@@ -121,16 +105,15 @@ void pm_power_state_exit_post_ops(struct pm_state_info info)
 
-	irq_unlock(ret);
+#ifdef CONFIG_DEBUG
+		/* Enable the Debug Module during STOP mode */
```
Please fix indentation

FRASTM replied:
done

FRASTM committed:
soc: arm: stm32l4 serie low power modes in debug config

The debug config will let the clocks active in STOP mode
at init.
Plus a minor fix in the Kconfig

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
plus a comment in the commit msg for the minor fix in the Kconfig

FRASTM committed:
boards: arm: stm32l476 nucleo board has low power states

The low power states 0∕1∕2 are added to the stm32l476rg
nucleo board, with default min residency values.
The substate-id is mapping the same Zephyr power state.
They correspond to the stm32l476 low power stop0/1/2 modes.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
soc: arm: stm32l4 low power substates

The low power states 0∕1∕2 are added to the stm32l4.
The substate-id is mapping the same Zephyr power state.
They correspond to the stm32l4 low power stop0/1/2 modes.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase and change the substate range

ABOSTM reviewed (changes requested):
Commit Message:
```
The low power states 0∕1∕2 are added to the stm32l476rg
nucleo board, with (default min residency values).
```
Sentence not finished: With what ? (or wrong parenthesis)
Otherwise LGTM

FRASTM commented:
rebase
changed info.substate_id in the pm_power_state_exit_post_ops function

ABOSTM reviewed (approved):
LGTM
